### ISSUE TITLE
Need to interface RASDR to SOAPY

### ISSUE THREAD

**Issue by fieldsde** [myriadrf/RASDR#22]
Interfacing to SOAPY would permit using with a large variety of SDRs.

The RASDR code remains the only SDR code specifically designed for amateur radio astronomy use.  It needs to be useable with more receivers, to achieve its true potential to the larger amateur radio astronomy community.

**Comment by bvacaliuc** [comment]
agree.  The challenge with the current setup is the integration of the control of the LMS6002D is integral to the way the RASDRview/RASDRproc software is organized.  A discussion on this w/r/t the RTL-SDR was started in #17, and Soapy was mentioned.

so between this ticket and the other one, getting this feature requires:

1. Converting the UI to talk only to the [Soapy SDR Client API](https://github.com/pothosware/SoapySDR/wiki#client-api)
2. Creating a [Soapy SDR Driver for the LMS6002D](https://github.com/pothosware/SoapySDR/wiki#driver-api)

This ticket asks for the first (and technically so does #17).  Let's make this ticket about supporting [SoapyLMS7](https://github.com/myriadrf/LimeSuite/tree/master/SoapyLMS7) as well as making a SoapyLMS6 and issuing a pull request upstream for it.


I think this request asks for RASDRproc to become a Soapy SDR client program and so to interface to other SDRs not just RASDR.

**Comment by fieldsde** [comment]
Bogdan,A very appropriate update.  But are now so far out on the coattails of development that we just have to jump ahead to more modern hardware and software tools?DavidOn Mar 21, 2024, at 02:27, Bogdan Vacaliuc ***@***.***> wrote:﻿
agree.  The challenge with the current setup is the integration of the control of the LMS6002D is integral to the way the RASDRview/RASDRproc software is organized.  A discussion on this w/r/t the RTL-SDR was started in #17, and Soapy was mentioned.
so between this ticket and the other one, getting this feature requires:

Converting the UI to talk only to the Soapy SDR Client API
Creating a Soapy SDR Driver for the LMS6002D

This ticket asks for the first (and technically so does #17).  Let's make this ticket about supporting SoapyLMS7 as well as making a SoapyLMS6 and issuing a pull request upstream for it.
I think this request asks for RASDRproc to become a Soapy SDR client program and so to interface to other SDRs not just RASDR.
